components/os/linux.go

```diff
@@ -18,25 +18,27 @@ func newLinuxOS(e config.Env, desc Descriptor, runner *command.Runner) OS {
 	case AmazonLinux, AmazonLinuxECS, CentOS:
 		// AL2 is YUM, AL2023 is DNF (but with yum compatibility)
 		os.packageManager = newYumManager(runner)
-		os.serviceManager = newSystemdServiceManager(e, runner)
 
```

Comment:
<!-- Suggestions propose improvements to the current subject. It’s important to be explicit and clear on what is being suggested and why it is an improvement. -->
**💬 suggestion**
I think you need to add the AmazonLinux2018 case or it would default to `panic`

```suggestion

	case AmazonLinux2018:
		os.PackageManager = new YumManager(runner)
		if desc.Version == AmazonLinux2018.Version {
			os.serviceManager = newSysvinitServiceManager(e, runner)
		} else {
			os.serviceManager = newSystemdServiceManager(e, runner)
		}
```

Reply:
The switch case is testing the flavors, and AmazonLinux2018 is an AmazonLinux flavor. FWIW I tested this whole PR locally and iterated until it worked